crates/red_knot_python_semantic/src/types.rs

```diff
@@ -120,6 +120,15 @@ fn symbol_by_id<'db>(db: &'db dyn Db, scope: ScopeId<'db>, symbol: ScopedSymbolI
 
 /// Shorthand for `symbol_by_id` that takes a symbol name instead of an ID.
 fn symbol<'db>(db: &'db dyn Db, scope: ScopeId<'db>, name: &str) -> Symbol<'db> {
+    // We don't need to check for `typing_extensions` here, because `typing_extensions.TYPE_CHECKING`
+    // is just a re-export of `typing.TYPE_CHECKING`.
+    if name == "TYPE_CHECKING"
+        && file_to_module(db, scope.file(db))
+            .is_some_and(|module| module.name().as_str() == "typing")
```

Comment:
I think it should be possible to compare ModuleNames with strings directly, without the `as_str()` call. You might need to borrow the l.h.s.?

https://github.com/astral-sh/ruff/blob/f52b1f4a4d076fc9e17d8341f2358a1522bd2f9e/crates/red_knot_python_semantic/src/module_name.rs#L200